engines/saga/font.cpp

```diff
@@ -672,105 +771,68 @@ void DefaultFont::outFont(const FontStyle &drawFont, const char *text, size_t co
+			// Advance tracking position
+			textPoint.x += _cjkFontWidth;
+			continue;
+		}
+		
+		// Translate character
+		if (_fontMapping == 0) {	// Check font mapping debug flag
+			// Default game behavior
+
+			// It seems that this font mapping causes problems with non-english
+			// versions of IHNM, so it has been changed to apply for ITE only.
+			// It doesn't make any difference for the English version of IHNM.
+			// Fixes bug #3405: "IHNM: Spanish font wrong".
+			if (!(flags & kFontDontmap) && _vm->getGameId() == GID_ITE) {
+				if (_vm->getLanguage() != Common::IT_ITA) {
+					c_code = translateChar(c_code);
+				} else {
+					// The in-game fonts of the Italian version should not be mapped.
+					// The ones in the intro are hardcoded and should be mapped normally.
+					if (_vm->_scene->isInIntro())
 						c_code = translateChar(c_code);
-					} else {
-						// The in-game fonts of the Italian version should not be mapped.
-						// The ones in the intro are hardcoded and should be mapped normally.
-						if (_vm->_scene->isInIntro())
-							c_code = translateChar(c_code);
-					}
 				}
-			} else if (_fontMapping == 1) {
-				// Force font mapping
-				c_code = translateChar(c_code);
-			} else {
-				// In all other cases, ignore font mapping
 			}
-			assert(c_code < FONT_CHARCOUNT);
+		} else if (_fontMapping == 1) {
+			// Force font mapping
+			c_code = translateChar(c_code);
+		} else {
+			// In all other cases, ignore font mapping
+		}
+		assert(c_code < FONT_CHARCOUNT);
 
-			// Check if character is defined
-			if ((drawFont.fontCharEntry[c_code].index == 0) && (c_code != FONT_FIRSTCHAR)) {
+		// Check if character is defined
+		if ((drawFont.fontCharEntry[c_code].index == 0) && (c_code != FONT_FIRSTCHAR)) {
 #if FONT_SHOWUNDEFINED
-				// A tab character appears in the IHNM demo instructions screen, so filter
-				// it out here
-				if (c_code == FONT_CH_SPACE || c_code == FONT_CH_TAB) {
-					textPoint.x += drawFont.fontCharEntry[c_code].tracking;
-					continue;
-				}
-				c_code = FONT_CH_QMARK;
-#else
-				// Character code is not defined, but advance tracking
-				// ( Not defined if offset is 0, except for 33 ('!') which
-				//   is defined )
+			// A tab character appears in the IHNM demo instructions screen, so filter
+			// it out here
+			if (c_code == FONT_CH_SPACE || c_code == FONT_CH_TAB) {
 				textPoint.x += drawFont.fontCharEntry[c_code].tracking;
 				continue;
-#endif
 			}
-
-			charWidth = drawFont.fontCharEntry[c_code].width;
-			charHeight = drawFont.header.charHeight;
-			rowLength = drawFont.header.rowLength;
-			bitmap = &drawFont.font[drawFont.fontCharEntry[c_code].index];
-			charTracking = drawFont.fontCharEntry[c_code].tracking;
+			c_code = FONT_CH_QMARK;
+#else
+			// Character code is not defined, but advance tracking
+			// ( Not defined if offset is 0, except for 33 ('!') which
+			//   is defined )
+			textPoint.x += drawFont.fontCharEntry[c_code].tracking;
```

Comment:
So this will be shown as a blank character, right? Is this what we want, or do we want to output ? in this case?

Reply:
Yes, as an empty char. It's easier to just skip, But putting '?' is also an option